dns.go

```diff
@@ -184,5 +185,21 @@ func getMapResponseDNSConfig(
 		dnsConfig = dnsConfigOrig
 	}
 
+	// Add device identifier to the nextdns resolvers, if there are any.
+	for _, resolver := range dnsConfig.Resolvers {
+		if strings.HasPrefix(resolver.Addr, "https://dns.nextdns.io/") {
```

Comment:
This would have similar concerns, I'm not too keen on having dedicated vendor stuff all over the code, but for this particular one, I could be convinced that having a generic interface/options/middelware system to override with could be appropriate, which is called from this function, but not in this function.